src/main/java/use_cases/add_edit_expenses_use_case/ExpenseUCI.java

```diff
@@ -181,7 +181,7 @@ private boolean checkHaveSameNameInList(ArrayList<Expense> expenseList, String n
      * (The user changes their response for the isRecurringExpense checkbox)
      * @return boolean suggesting there is a need to update Session.recurringExpenseData
      */
-    private boolean changeInRecurringInfo() {
+    private boolean changeInRecurringInfo(ExpenseID expenseID) {
```

Comment:
While trying to review I kept getting confused of what this method was doing, but after looking at it I think it could be written in a clearer way, especially its JavaDoc. I changed it as an example

```java
    /**
     * Returns whether there's a difference in the recurring status of the old expense and the edited expense.
     * This could mean that the expense wasn't recurring and now the user is trying to make it recurring,
     * or vice versa.
     *
     * @return a boolean describing whether there was a change in the recurring status of the expense
     */
    private boolean changeInRecurringInfo(ExpenseID expenseID) {
        boolean isOldExpenseRecurring;
        try {
            findExpense(expenseID.getSession().getRecurData(), expenseID.getOldExpense());
            isOldExpenseRecurring = false;
        } catch (NoSuchElementException e) {
            isOldExpenseRecurring = true;
        }

        return isOldExpenseRecurring != expenseID.getIsRecurringExpense();
    }
```

Reply:
I admit it is a little confusing, and I never thought of assigning a variable. If you don't mind, I will change it to your version!

Reply:
I think the logic in your implementation is flipped. I believe the following will be correct!
```
 private boolean changeInRecurringInfo(ExpenseID expenseID) {
        boolean isOldExpenseRecurring;
        try {
            findExpense(expenseID.getSession().getRecurData(), expenseID.getOldExpense());
            isOldExpenseRecurring = true;
        } catch (NoSuchElementException e) {
            isOldExpenseRecurring = false;
        }
        return isOldExpenseRecurring != expenseID.getIsRecurringExpense();
    }
```